qiskit/opflow/gradients/circuit_qfis/overlap_block_diag.py

```diff
@@ -54,7 +55,7 @@ def convert(self,
         Raises:
             NotImplementedError: If ``operator`` is neither ``CircuitOp`` nor ``CircuitStateFn``.
         """
-        if not isinstance(operator, (CircuitOp, CircuitStateFn)):
```

Comment:
Why is this changed?

Reply:
This change correspond to the state on master:
```if not isinstance(operator, (CircuitStateFn)):```
But it is not consistent with the doc-string and error message, so we need to modify one of those...

Reply:
I added `CircuitOp` again, because both `overlap_diag` methods use only `circuit = operator.primitive`